### PR TITLE
Ruby setup

### PR DESCRIPTION
**Pull request by lsf37**
- update Gemfile.lock and bring in line with pinned Ruby version
- set up correct Ruby version in CI
- cache Ruby deps in CI
- describe `rbenv` setup in README
- reorg README to make install/build easier

### CODE REVIEW

**Comment by lsf37** [comment]
I'm not entirely sure why I had to downgrade sass-embedded from `1.54` to `1.53`, but I got dependency conflicts for anything > 1.53 and it's not like we're using this on untrusted input or are currently hitting bugs. Longer term it'd be nice to upgrade the Ruby version, but that's more time than I'm willing to spend today.

**Comment by lsf37** [comment]
Tested with:
- Darwin Intel, Darwin Apple Silicon
- Linux x64 and arm64 via Docker

**Review by Indanz — APPROVED** [review]
My main concern is that you seem to be manually editing Gemfile.lock. You should never do that, all changes should come via requests in Gemfile or bundler update or similar.

The other changes seem fine.

**Comment by lsf37** [comment]
> My main concern is that you seem to be manually editing Gemfile.lock. You should never do that, all changes should come via requests in Gemfile or bundler update or similar.

The commit is the result of `bundle update`, with some versions constrained in `Gemfile` (in the same commit) and the Ruby version (and therefore initial bundler version) pinned in `.ruby-version`. This should be fully stable.

File: README.md [hunk 1]
```diff
@@ -3,83 +3,141 @@
...
-#### Building the site
+#### Python
 
-- ruby-bundler must be installed
+The build is tested with Python 3.9. More recent versions are likely to work as
+well. After installing Python via e.g. `homebrew` or `apt`, you can install the
+Python build dependencies with:
+
+```sh
+pip3 install --user camkes-deps
```
[the remark this answers is not in the document]
Comment (lsf37) [reply README.md:73]: It is somewhat surprising yes. AFAICT these are mostly for the tutorial solutions.

File: README.md [hunk 2]
```diff
@@ -3,83 +3,141 @@
...
-One way we use this is to show data generated by rules in the Makefile and saved in \_data/generated.yml.
-If you want to serve the site locally in production mode there are make rules for this.  You will need
-to call the make rule to generate the \_data/generated.yml also.
+One way we use this is to show data generated by rules in the Makefile and saved
+in `\_data/generated.yml`. If you want to serve the site locally in production
+mode there are `make` rules for this.  You will need to call the make rule to
+generate the `\_data/generated.yml` also.
 
-## Quick compliance, style and formatting checks
+## Diagnosing a problem
```
[the remark this answers is not in the document]
Comment (lsf37) [reply README.md:111]: True, I kept the old heading, but it doesn't really fit. Will change.

**Comment by lsf37** [comment]
Rebased, addressed review feedback and documented in the commit message how Gemfile.lock was updated for the future.

**Comment by lsf37** [comment]
That dependency caching for Ruby is quite nice. The build is much faster now.

**Pull request merged**